python/pyarrow/array.pxi

```diff
@@ -635,7 +635,7 @@ cdef class Array(_PandasConvertible):
 
         Parameters
         ----------
-        sequence : ndarray, Inded Series
+        sequence : ndarray, Indexed Series
```

Comment:
This seems odd. Change this to simply `sequence: ndarray, pandas.Series, or array-like`